cmd/routed-eni-cni-plugin/driver/driver.go

```diff
@@ -43,6 +44,8 @@ const (
 	fromContainerRulePriority = 1536
 	// Main routing table number
 	mainRouteTable = unix.RT_TABLE_MAIN
+
+	WAIT_INTERVAL = 50 * time.Millisecond
```

Comment:
Good find ! Not sure if we plan to support anything beyond AL2 for initial cut.  But it might be worth checking if this delay holds good in other distributions as well. 

Side note : Going forward we will have multiple eth attachments on single pod with 5G to separate out different flows.  Having this delay as configurable option would help until we characterize the actual number for different use cases.

Reply:
So, the total wait time is actually `10s`. WAIT_INTERVAL` is essentially how long we wait before checking the status again. I'm assuming `10s` might be long enough and the function in `ip` package that most of the CNI plugins rely on is capping it @`10s` as well. I see that it usually takes between 1-2s in my testing but if we do run in to a specific requirement/use-case, I guess we can definitely consider making the upper bound configurable.

Reply:
Fwiw, I don't think it needs to be configurable - we should be able to just 'wait long enough' for every use case, and there's no benefit from timing out and aborting aggressively.

Re other distros: It would be odd to pick something wildly different from the current Linux kernel default values, and I expect the delay will always be around the few-seconds mark.  One alternative here is that we either use 'optimistic DAD' which allows userspace to use the address for _some_ purposes while it is still tentative.  A better alternative is to just disable DAD altogether on veth interfaces, because we control both ends anyway so there are no surprises here.  Meh, at best it gains 1-2s, and we can come back to this later.  Even if we disable DAD on veth, we're still going to want this function at some point for "real" network interfaces (eg: trunk, EFA, ENI+ipvlan).

We could also remove the above timer by using netlink events rather than polling (see [AddrSubscribe](https://github.com/vishvananda/netlink/blob/53455394aa2b7ada7a308d0328747acdd05e91d5/addr_linux.go#L296-L300)).  Again, meh, we can come back to this if this 50ms poll ever becomes an issue.